core/src/main/java/org/elasticsearch/discovery/zen/ZenDiscovery.java

```diff
@@ -353,6 +358,7 @@ public boolean joiningCluster() {
     private void innerJoinCluster() {
         DiscoveryNode masterNode = null;
         final Thread currentThread = Thread.currentThread();
+        nodeJoinController.startAccumulatingJoins();
```

Comment:
why are we going to accumulate join request while we don't know that we are going to become master master?